src/components/AvatarWithImagePicker.js

```diff
@@ -51,17 +61,97 @@ const defaultProps = {
     style: [],
     DefaultAvatar: () => {},
     isUsingDefaultAvatar: false,
+    isUploading: false,
+    size: CONST.AVATAR_SIZE.DEFAULT,
 };
 
 class AvatarWithImagePicker extends React.Component {
     constructor(props) {
         super(props);
 
+        this.rotate = new Animated.Value(0);
+        this.scale = new Animated.Value(1);
+        this.startRotation = this.startRotation.bind(this);
+        this.startUploadIndicator = this.startUploadIndicator.bind(this);
+        this.stopUploadIndicator = this.stopUploadIndicator.bind(this);
+
         this.state = {
             isMenuVisible: false,
         };
     }
 
+    componentDidMount() {
+        if (this.props.isUploading) {
+            this.startUploadIndicator();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!prevProps.isUploading && this.props.isUploading) {
+            this.startUploadIndicator();
+        } else if (prevProps.isUploading && !this.props.isUploading) {
+            this.stopUploadIndicator();
+        }
+    }
+
+    componentWillUnmount() {
+        this.stopUploadIndicator();
+    }
+
+    /**
+     * We need to manually loop the animations as `useNativeDriver` does not work well with Animated.loop.
+     *
+     * @memberof AvatarWithImagePicker
+     */
+    startRotation() {
+        this.rotate.setValue(0);
+        Animated.timing(this.rotate, {
```

Comment:
Rather than using recursion here, it's better to use `Animated.loop` because it won't block the JS thread to make the recursive call:

```javascript
Animated.loop(
    Animated.timing(this.rotate, {
        toValue: 1,
        duration: 2000,
        easing: Easing.linear,
        isInteraction: false,
        useNativeDriver: true,
    }),
).start();
```

Reply:
i.e: the animation will restart immediately in the UI thread, instead of:

1. Sending a message through the bridge back to JS to say the animation is done
1. JS thread is blocked to execute the callback
1. Callback sends a message back through the bridge to UI thread to restart the animation.

Reply:
Thanks for the clarification. I haven't worked with Animation earlier. It seems though we're using the same thing in `AvatarWithIndicator`. What should we do about that?

Reply:
Updated and works great. Thanks for the suggestion. 🙌